src/actions/profilesActions.ts

```diff
@@ -155,71 +244,56 @@ export function updateProfile(accountAddress: string, name: string, description:
       Description: description,
     });
 
-    dispatch(showNotification(NotificationStatus.Success, "Profile data saved to 3box"));
+    dispatch(showNotification(NotificationStatus.Success, i18next.t("3BoxProfileSuccess")));
     return true;
   };
 }
 
-export type FollowItemAction = IAsyncAction<"FOLLOW_ITEM", { accountAddress: string }, { type: FollowType; id: string; isFollowing: boolean}>;
+export type FollowItemAction = IAsyncAction<"FOLLOW_ITEM", { accountAddress: string }, IFollowingPayload>;
 
 export function toggleFollow(accountAddress: string, type: FollowType, id: string) {
   return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState): Promise<boolean> => {
-    const state = _getState();
-    let threeBox;
     let threeBoxSpace;
+    let isFollowing = true;
 
     try {
-      if (state.profiles.threeBox) {
-        threeBox = state.profiles.threeBox;
-      } else {
-        const web3Provider = await getWeb3Provider();
-        threeBox = await Box.openBox(accountAddress, web3Provider);
-      }
+      threeBoxSpace = (await get3Box(accountAddress, dispatch, _getState(), true)).threeBoxSpace;
 
-      await threeBox.syncDone;
+      let follows = await threeBoxSpace.public.get("follows");
+      if (!follows) {
+        follows = {
+          daos: [],
+          proposals: [],
+          users: [],
+        };
+      }
+      if (!follows[type]) {
+        follows[type] = [];
+      }
 
-      if (state.profiles.threeBoxSpace) {
-        threeBoxSpace = state.profiles.threeBoxSpace;
+      if (follows[type].includes(id)) {
+        follows[type] = arrayRemove(follows[type], id);
+        isFollowing = false;
       } else {
-        threeBoxSpace = await threeBox.openSpace("DAOstack");
+        follows[type].push(id);
       }
-      await threeBoxSpace.syncDone;
-    } catch (e) {
-      dispatch(showNotification(NotificationStatus.Failure, `Failed to connect to 3box: ${e.message}`));
-      return false;
-    }
 
-    let follows = await threeBoxSpace.public.get("follows");
-    if (!follows) {
-      follows = {
-        daos: [],
-        proposals: [],
-        users: [],
-      };
-    }
-    if (!follows[type]) {
-      follows[type] = [];
-    }
+      // TODO: check success?
+      await threeBoxSpace.public.set("follows", follows);
 
-    let isFollowing = true;
+      dispatch({
+        type: ActionTypes.FOLLOW_ITEM,
+        sequence: AsyncActionSequence.Success,
+        meta: { accountAddress },
+        payload: { type, id, isFollowing },
+      } as FollowItemAction);
 
-    if (follows[type].includes(id)) {
-      follows[type] = arrayRemove(follows[type], id);
-      isFollowing = false;
-    } else {
-      follows[type].push(id);
+    } catch (e) {
+      dispatch(showNotification(NotificationStatus.Failure, `Failed to connect to 3box: ${e.message}`));
```

Comment:
is this can be moved to the string place ?

Reply:
That is old code.  I have not tasked myself in this PR with moving every single string into i18.